src/vm/dllimport.cpp

```diff
@@ -6574,7 +6611,7 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
             hmod = PAL_RegisterLibraryDirect(hmod, wszLibName);
```

Comment:
(unrelated to the current change) I am wondering whether we need the `PAL_RegisterLibraryDirect` at all after your refactoring. Looks like we may be able to switch the rest of this to use the `NATIVE_LIBRARY_HANDLE` directly as well.

Reply:
@jkotas  and I talked about this. @jkotas said that we can do away with the call to DllMain on Unix systems, thus making the PAL abstraction on native handles unnecessary. This will be addressed in a separate change. CC: @janvorli

Reply:
https://github.com/dotnet/coreclr/issues/21009 Tracks this.